validation/src/main/scala/hmda/validation/rules/lar/validity/V290.scala

```diff
@@ -0,0 +1,38 @@
+package hmda.validation.rules.lar.validity
+
+import hmda.model.census.CBSATractLookup
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.{ Failure, Result }
+import hmda.validation.rules.EditCheck
+
+object V290 extends EditCheck[LoanApplicationRegister] {
+
+  val cbsaTracts = CBSATractLookup.values
+
+  val validCombinations = cbsaTracts.map { cbsa =>
+    (cbsa.state, cbsa.county, cbsa.geoidMsa)
+  }
+
+  override def name: String = "V290"
+
+  override def apply(input: LoanApplicationRegister): Result = {
+    val msa = msaCode(input.geography.msa)
+    val state = input.geography.state
+    val county = input.geography.county
+
+    val combination = (state, county, msa)
+
+    when(msa not equalTo("NA")) {
+      combination is containedIn(validCombinations)
+    }
+  }
+
+  private def msaCode(code: String): String = {
+    val md = cbsaTracts.filter(m => m.metDivFp == code)
```

Comment:
I am not concerned about this at this point. I would be surprised if this became our bottleneck. For some context, the HMDA pilot had this loaded into a `MongoDB` database and this edit check required a REST API call. That implementation was fast enough for 98% of HMDA filers according to their tests. Avoiding the network hop is already making this much faster. The tradeoff is memory of course. In any case I agree that we will need to measure when the system is deployed (in a cluster)